### PR TITLE
Add aarch64 for linux

### PR DESCRIPTION
**Pull request by keitherskine**
Update the CI pipeline to build wheels for ARM64 Linux processors.  As requested in #837 .

Whilst we're here, remove the `push`/`pull_request` triggers for this workflow.  We don't need to build wheels every time a commit is pushed.  Keep `workflow_dispatch` so that this workflow can be manually triggered at release time.

### CODE REVIEW

**Comment by keitherskine** [comment]
Had to make a couple of changes to get the tests to pass.

- In Github Actions, `ubuntu-latest` is now 22.04 which isn't available for Python 3.6, so pinned the runner to 20.04 for the unit tests.
- In AppVeyor, for some reason I still don't understand, downloading MySQL drivers no longer works with the standard PowerShell command `Invoke-WebRequest` (getting 403's).  Downloading drivers for SQL Server and Postgres drivers is still OK, just not MySQL.  Weird.  Using `curl` seems to work so implemented that instead.  `curl` also has the advantage of having the `--verbose` option which `Invoke-WebRequest` doesn't have.

As well as these two fixes, I've added various other tweaks to make the AppVeyor log output better, and to make sure the AppVeyor cache of driver msi files for the jobs is kept up-to-date and correct.

This PR should be good to review now.

File: appveyor/install.ps1 [hunk 1]
```diff
@@ -1,24 +1,33 @@
 # check that all the required ODBC drivers are available, and install any that are missing
 
 Function DownloadFileFromUrl ($url, $file_path) {
+    $curl_params = "-f -sS -L -o `"$file_path`" `"$url`""
+    If (${env:APVYR_VERBOSE} -eq "true") {
```
Comment (AlexandreArpin) [inline appveyor/install.ps1:5]: nitpick:  all the other code uses the lowercase form of built-in functions.

```suggestion
    if (${env:APVYR_VERBOSE} -eq "true") {
```

**Review by gordthompson — APPROVED** [review]
lgtm

File: appveyor/install.ps1 (hunk 1 above)
Comment (gordthompson) [reply appveyor/install.ps1:5]: +1 - may as well be consistent

Comment (keitherskine) [reply appveyor/install.ps1]: Will do.

**Pull request merged**